src/mono/mono/metadata/class-init.c

```diff
@@ -1092,6 +1077,19 @@ mono_class_create_bounded_array (MonoClass *eclass, guint32 rank, gboolean bound
 		g_free (name);
 	}
 
+	nsize = strlen (eclass->name);
+	name = (char *)g_malloc (nsize + 2 + rank + 1);
+	memcpy (name, eclass->name, nsize);
+	name [nsize] = '[';
+	if (rank > 1)
+		memset (name + nsize + 1, ',', rank - 1);
+	if (bounded)
+		name [nsize + rank] = '*';
+	name [nsize + rank + bounded] = ']';
+	name [nsize + rank + bounded + 1] = 0;
+	klass->name = image_set ? mono_image_set_strdup (image_set, name) : mono_image_strdup (image, name);
+	g_free (name);
```

Comment:
The code in the `if (rank > 32)` block immediately above this code uses `klass->name` (via `mono_type_get_full_name` which is not initialized yet (because that code moved down).
So I think we need to put something else in the error message.  Maybe just `"Cannot create array of %s of rank %d", eklass_name, rank` where `eklass_name` is `mono_type_get_full_name (eklass)`

Also put all this code here that sets up `name`in the `else` branch so that we don't, create a huge `"T[,,,,...,,,,]"` string.